### DISCUSSION TITLE
Server Resource Pack Spoofing feature

### DISCUSSION THREAD

**Discussion by RoyDubnium** [maruohon/tweakeroo#297] [Ideas]
I have a suggestion for a tweak or standalone mod that allows the client to pretend to load a server resource pack, while in fact it does not, like how it works in [Minecraft Console client](https://github.com/MCCTeam/Minecraft-Console-Client/issues/91). This would help if a server requires the use of a custom resource pack, but it has to load it every single time. Or, alternatively, it could just load the server resource pack and KEEP it loaded even when you leave.

**Comment by maruohon** [comment]
I have to say I'm not at all familiar with how these work... Is it actually possible for a server to *require* you to use a resource pack? I always get the dialog asking if I want to use server resource packs when I join to a new server for the first time(?) and I always just assumed if I answer `Never` then it would just not use one and not ask again. Is this not how it works?

**Reply by RoyDubnium** [reply]
From [this issue](https://github.com/MCCTeam/Minecraft-Console-Client/issues/91), I have gathered that your client sends a confirmation packet back to the server, both when it starts and when it finishes loading the pack.  This allows them to require the use of a resource pack. [This script](https://github.com/MCCTeam/Minecraft-Console-Client/commit/344749ead2f070ba0c4b761a2e614d4030dcf9bd) shows how Minecraft console client handles this.

**Reply by RoyDubnium** [reply]
Perhaps you could just send these packets every time the client connects, not just if the pack request (packet ID 0x48) is recieved.

SendPacket(0x19, concatBytes(getVarInt(hash.Length), Encoding.UTF8.GetBytes(hash), getVarInt(3)));
SendPacket(0x19, concatBytes(getVarInt(hash.Length), Encoding.UTF8.GetBytes(hash), getVarInt(0)));

**Reply by RoyDubnium** [reply]
ok now im just confused
